12 luna

MadeddFriend opened the pull request.

MadeddFriend committed:
старт проекта "12-luna"

MadeddFriend committed:
Решение задачи

convertAndReverse - функция удаляет пробелы и "-" между цифрами и преобразует каждую в отдельные элемент массива, также разворачивает его для того, чтобы упростить логику и возвращает массив с числами.

checkLuhn - функция с помощью метода reduce проверяет числа согласно логике алгоритма, с учетом того, что массив розвернут.

Levitinn commented on app.js:
```diff
@@ -0,0 +1,35 @@
+
+const convertAndReverse = str => {
+	let res
+	if (str.includes('-'))
```
слишком много циклов. ты можешь за один проход по строке исправить и "-" и пробелы

MadeddFriend committed:
Внёс изменения

Исправил работу функции convertAndReverse, заменил повторяющиеся части кода и циклы на метод replace + регулярное выражение /[-\s]/g
